index.js

```diff
@@ -122,36 +127,52 @@ async function openTelemetryPlugin (fastify, opts = {}) {
   }
 
   async function onError (request, reply, error) {
-    if (shouldIgnoreRoute(request.url, request.method)) return
-
     const activeContext = getContext(request)
     const span = trace.getSpan(activeContext)
     span.setAttributes(formatSpanAttributes.error(error))
   }
 
-  fastify.addHook('onRequest', onRequest)
-  fastify.addHook('onResponse', onResponse)
-  fastify.addHook('onError', onError)
-
-  if (wrapRoutes) {
-    function wrapRoute (routeHandler) {
-      return async function (request, ...args) {
-        const reqContext = getContext(request)
-        return context.with(reqContext, routeHandler.bind(this, request, ...args))
+  function executeIfNotWrappedOrNotIgnoredRoute (handler) {
+    return async function (request, ...params) {
+      if (!shouldWrapRoute(request.routerPath) && !shouldIgnoreRoute(request.url, request.method)) {
+        return handler(request, ...params)
       }
     }
+  }
+
+  fastify.addHook('onRequest', onRequest)
```

Comment:
Could we use the same pattern as `onResponse` and on `onError` here as well? This way we can ensure that our `onRequest` hook always get called first

```js
fastify.addHook('onRequest', executeIfNotWrappedOrNotIgnoredRoute(onRequest))

// ...

routeOpts.onRequest = [onRequest, ...normalizeHooks(routeOpts.onRequest)]
```